src/hooks/limitOrders/useGelatoLimitOrdersHistory.ts

```diff
@@ -1,168 +1,114 @@
-                const orderCancelled = pendingOrdersLS
-                  .filter((pendingOrder) => pendingOrder.status === 'cancelled')
-                  .find((pendingOrder) => pendingOrder.id.toLowerCase() === order.id.toLowerCase())
-                return !orderCancelled
-              })
-              .sort(newOrdersFirst),
-            pending: pendingOrdersLS.filter((order) => order.status === 'open').sort(newOrdersFirst),
-          })
-        })
-        .catch((e) => {
-          console.error('Error fetching open orders from subgraph', e)
-          const openOrdersLS = getLSOrders(chainId, account).filter((order) => order.status === 'open')
-
-          const pendingOrdersLS = getLSOrders(chainId, account, true)
-
-          setOpenOrders({
-            confirmed: openOrdersLS
-              .filter((order: Order) => {
-                const orderCancelled = pendingOrdersLS
-                  .filter((pendingOrder) => pendingOrder.status === 'cancelled')
-                  .find((pendingOrder) => pendingOrder.id.toLowerCase() === order.id.toLowerCase())
-                return !orderCancelled
-              })
-              .sort(newOrdersFirst),
-            pending: pendingOrdersLS.filter((order) => order.status === 'open').sort(newOrdersFirst),
-          })
-        })
-  }, [gelatoLimitOrders, account, chainId, includeOrdersWithNullHandler])
-
-  const fetchCancelledOrders = useCallback(() => {
-    if (gelatoLimitOrders && account && chainId)
-      gelatoLimitOrders
-        .getCancelledOrders(account.toLowerCase(), includeOrdersWithNullHandler)
-        .then(async (orders) => {
-          const ordersLS = getLSOrders(chainId, account)
-
-          orders.forEach((order: Order) => {
-            const orderExists = ordersLS.find((confOrder) => confOrder.id.toLowerCase() === order.id.toLowerCase())
-            if (!orderExists || (orderExists && Number(orderExists.updatedAt) < Number(order.updatedAt))) {
-              saveOrder(chainId, account, order)
-            }
-          })
+export function useGelatoOpenLimitOrders(): Order[] {
+  const { account, chainId } = useActiveWeb3React()
 
-          const cancelledOrdersLS = getLSOrders(chainId, account).filter((order) => order.status === 'cancelled')
+  const gelatoLimitOrders = useGelatoLimitOrdersLib()
 
-          const pendingCancelledOrdersLS = getLSOrders(chainId, account, true).filter(
-            (order) => order.status === 'cancelled',
-          )
+  const { data } = useSWR(
+    gelatoLimitOrders && account && chainId ? ['gelato', 'openOrders'] : null,
+    async () => {
+      try {
+        const orders = await gelatoLimitOrders.getOpenOrders(account.toLowerCase(), false)
 
-          setCancelledOrders({
-            confirmed: cancelledOrdersLS.sort(newOrdersFirst),
-            pending: pendingCancelledOrdersLS.sort(newOrdersFirst),
-          })
+        syncOrderToLocalStorage({
+          orders,
+          chainId,
+          account,
```

Comment:
We can simple use swr middleware + localstorage 0xa7A01B93B889ff0639d5ec02914A77529924a46F

Reply:
I think @Chef-Cheems  is working on a new local storage mechanism. Waiting for his approach andI and I could refactor it.